eq_cir_proxy_service/routers/instrument_router.py

```diff
@@ -0,0 +1,47 @@
+"""Module defines the instrument router for handling requests related to instruments in the EQ CIR Proxy Service."""
+
+from uuid import UUID
+
+from fastapi import APIRouter, HTTPException, Path, Query
+
+from eq_cir_proxy_service.config.logging_config import logging
+from eq_cir_proxy_service.exception import exception_messages
+from eq_cir_proxy_service.services.schema import schema_retrieval_service
+from eq_cir_proxy_service.services.validators.request_validator import (
+    validate_version,
+)
+from eq_cir_proxy_service.types.custom_types import Schema
+
+router = APIRouter()
+logger = logging.getLogger(__name__)
+INSTRUMENT_ID_PATH = Path(..., description="UUIDv4 of the instrument")
+
+
+@router.get("/instrument/{instrument_id}")
+async def get_instrument_by_uuid(
+    instrument_id: UUID = INSTRUMENT_ID_PATH,
+    version: str = Query(default=None, description="Optional version of the instrument"),
+) -> Schema:
+    """Retrieve an instrument by its UUID and optional version."""
+    logger.info("Receiving the instrument id...")
+    logger.debug("Received instrument id: %s", instrument_id)
```

Comment:
Should we be doing some kind of validation on the instrument_id, e.g. that it is a valid uuid? I think at the moment we just get a 404 with an invalid uuid which could potentially be confusing.

Reply:
I originally had this during development, but as the type for instrument_id is as set UUID, FastAPI automatically checks and validates if the value is a UUID, and triggers the appropriate error if needed. You can see it if you just put say '12' as the instrument_id.